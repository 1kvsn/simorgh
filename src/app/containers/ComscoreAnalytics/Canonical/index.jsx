/* eslint-disable jsx-a11y/alt-text */
import React, { useContext } from 'react';
import { Helmet } from 'react-helmet';
import { UserContext } from '#contexts/UserContext';

const CanonicalComscoreAnalytics = () => {
  const { cookiePolicy } = useContext(UserContext);

  let csUcfr = 0;
  if (cookiePolicy) {
    csUcfr = cookiePolicy.slice(-1);
  }

  return (
    <Helmet>
      <script
        async
        type="text/javascript"
        src={`${process.env.SIMORGH_PUBLIC_STATIC_ASSETS_ORIGIN}/simorgh/public/comscore.js`}
      />
      <noscript>
        {`<img src="https://sb.scorecardresearch.com/p?c1=2&c2=17986528&cs_ucfr=${csUcfr}&cv=2.0&cj=1" />`}
      </noscript>
    </Helmet>
  );
};

export default CanonicalComscoreAnalytics;
