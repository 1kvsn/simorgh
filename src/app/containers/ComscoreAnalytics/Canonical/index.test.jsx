import React from 'react';
import { render, waitFor } from '@testing-library/react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import ContextWrap from '../testHelper';
import ComscoreAnalytics from '..';

describe('Canonical Comscore Analytics ', () => {
  const toggleState = {
    comscoreAnalytics: {
      enabled: true,
    },
  };

  describe('Assertions', () => {
    it('should render null when not on client', async () => {
      const { container } = render(
        <ContextWrap
          platform="canonical"
          pageType="article"
          origin="bbc.com"
          toggleState={toggleState}
        >
          <ComscoreAnalytics />
        </ContextWrap>,
      );

      expect(container).toBeEmptyDOMElement();
    });

    it('should render canonical comscore with noscript and script element', async () => {
      const noScriptContent = `<img src="https://sb.scorecardresearch.com/p?c1=2&c2=17986528&cs_ucfr=0&cv=2.0&cj=1" />`;

      render(
        <ContextWrap
          platform="canonical"
          pageType="article"
          origin="bbc.com"
          toggleState={toggleState}
        >
          <ComscoreAnalytics />
        </ContextWrap>,
      );
      await waitFor(() => {
        const noScriptEl = document.querySelector('noscript');
        expect(noScriptEl).toBeInTheDocument();
        expect(noScriptEl.textContent).toEqual(noScriptContent);

        const scriptEl = document.querySelector('script');
        expect(scriptEl).toBeInTheDocument();
        expect(scriptEl).toHaveAttribute('async');
        expect(scriptEl.src).toEqual(
          'http://localhost:7080/static/js/comscore-1.0.js',
        );
      });
    });
  });

  describe('Snapshots', () => {
    shouldMatchSnapshot(
      'should render comscore script when on canonical and toggle is enabled',
      <ContextWrap
        platform="canonical"
        pageType="article"
        origin="bbc.com"
        toggleState={toggleState}
      >
        <ComscoreAnalytics />
      </ContextWrap>,
    );
  });
});
