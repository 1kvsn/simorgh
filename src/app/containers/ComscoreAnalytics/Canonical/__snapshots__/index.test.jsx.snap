// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canonical Comscore Analytics  Snapshots should render comscore script when on canonical and toggle is enabled 1`] = `
<html>
  <head>
    <noscript>
      &lt;img src="https://sb.scorecardresearch.com/p?c1=2&c2=17986528&cs_ucfr=0&cv=2.0&cj=1" /&gt;
    </noscript>
    <script
      async="true"
      src="http://localhost:7080/static/js/comscore-1.0.js"
      type="text/javascript"
    />
  </head>
  <body>
    <div />
  </body>
</html>
`;
